domehunter/__init__.py

```diff
@@ -86,70 +230,291 @@ def is_home(self):
+        if self.dome_az is None:
+            self.calibrate()
+        return self.dome_az
+
+    def GotoAz(self, az):
+        "Send Dome to a requested Azimuth position."
+        if self.dome_az is None:
+            self.calibrate()
+        delta_az = az - self.dome_az
+
+        # determine whether CW or CCW gives the short path to desired az
+        if abs(delta_az) > 180:
+            if delta_az > 0:
+                delta_az -= 360
+            else:
+                delta_az += 360
+
+        # if updated delta_az is positive, direction is CW
+        if delta_az > 0:
+            # converted delta_az to equivilant in encoder ticks
+            ticks = self._az_to_ticks(delta_az)
+            target_position = self.encoder_count + ticks
+            self._move_cw()
+            # wait until encoder count matches desired delta az
+            while self.encoder_count < target_position:
+                if self.testing:
+                    # if testing simulate a tick for every cycle of while loop
+                    self._simulate_ticks(num_ticks=1)
+                pass
+            self._stop_moving()
+            # compare original count to current just in case we got more ticks
+            # than we asked for
+            old_encoder_count = target_position - ticks
+            # update dome_az based on the actual number of ticks counted
+            self.dome_az += self._ticks_to_az(
+                self.encoder_count - old_encoder_count)
+            # take mod360 of dome_az to keep 0 <= dome_az < 360
+            self.dome_az %= 360
+            # update encoder_count to match the dome_az
+            self.encoder_count = self._az_to_ticks(self.dome_az)
+            pass
+
+        # if updated delta_az is negative, direction is CCW
+        if delta_az < 0:
+            # converted delta_az to equivilant in encoder ticks
+            ticks = self._az_to_ticks(delta_az)
+            target_position = self.encoder_count + ticks
+            self._move_ccw()
+            # wait until encoder count matches desired delta az
+            while self.encoder_count >= target_position:
+                if self.testing:
+                    # if testing simulate a tick for every cycle of while loop
+                    self._simulate_ticks(num_ticks=1)
+                pass
+            self._stop_moving()
+            # compare original count to current, just in case we got more ticks
+            # than we asked for
+            old_encoder_count = target_position - ticks
+            # update dome_az based on the actual number of ticks counted
+            self.dome_az += self._ticks_to_az(
+                self.encoder_count - old_encoder_count)
+            # take mod360 of dome_az to keep 0 <= dome_az < 360
+            self.dome_az %= 360
+            # update encoder_count to match the dome_az
+            self.encoder_count = self._az_to_ticks(self.dome_az)
+            pass
         pass
```

Comment:
If requested Az is `0`, I guess nothing happens, which is fine. What if someone typos a `3600` I guess it will spin a lot? Maybe just check if its a valid range?